pkg/gardenlet/controller/shoot/shoot_control_prepare_migration.go

```diff
@@ -0,0 +1,386 @@
+		})
+		wailtUntilBackupEntryInGardenReconciled = g.Add(flow.Task{
+			Name:         "Waiting until the backup entry has been reconciled",
+			Fn:           flow.TaskFn(botanist.WaitUntilBackupEntryInGardenReconciled).DoIf(nonTerminatingNamespace),
+			Dependencies: flow.NewTaskIDs(deployBackupEntryInGarden),
+		})
+		deployETCD = g.Add(flow.Task{
+			Name:         "Deploying main and events etcd",
+			Fn:           flow.TaskFn(botanist.DeployETCD).RetryUntilTimeout(defaultInterval, defaultTimeout).DoIf(nonTerminatingNamespace),
+			Dependencies: flow.NewTaskIDs(deploySecrets, wailtUntilBackupEntryInGardenReconciled),
+		})
+		waitUntilEtcdReady = g.Add(flow.Task{
+			Name:         "Waiting until main and event etcd report readiness",
+			Fn:           flow.TaskFn(botanist.WaitUntilEtcdReady).DoIf(nonTerminatingNamespace),
+			Dependencies: flow.NewTaskIDs(deployETCD),
+		})
+		annotateExtensionCRDsForMigration = g.Add(flow.Task{
+			Name: "Annotate Extensions with operation - migration",
+			Fn:   botanist.AnnotateExtensionCRDsForMigration,
+		})
+		prepareControlPlaneDeployments = g.Add(flow.Task{
+			Name:         "Prepare Control Plane Deployments in Shoot's namespace",
+			Fn:           flow.TaskFn(botanist.PrepareControlPlaneDeployments).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(shootHibernated),
+			Dependencies: flow.NewTaskIDs(waitUntilEtcdReady),
+		})
+		waitUntilAPIServerIsDown = g.Add(flow.Task{
+			Name:         "Wait for Kubernetes API Server to be scaled to zero",
+			Fn:           flow.TaskFn(botanist.WaitUntilKubeAPIServerScalesDown).SkipIf(shootHibernated),
+			Dependencies: flow.NewTaskIDs(prepareControlPlaneDeployments),
+		})
+		waitUntilResourceManagerIsDown = g.Add(flow.Task{
+			Name:         "Wait for gardener-resource-manager to be scaled to zero",
+			Fn:           flow.TaskFn(botanist.WaitUntilGardenerResourceManagerScalesDown).SkipIf(shootHibernated),
+			Dependencies: flow.NewTaskIDs(prepareControlPlaneDeployments),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying ingress DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyIngressDNSRecord).DoIf(dnsEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerIsDown),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying external domain DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyExternalDomainDNSRecord).DoIf(dnsEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerIsDown),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying internal domain DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyInternalDomainDNSRecord).DoIf(dnsEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerIsDown),
+		})
+		createETCDSnapshot = g.Add(flow.Task{
+			Name:         "Create ETCD Snapshot",
+			Fn:           flow.TaskFn(botanist.CreateETCDSnapshot).DoIf(false),
+			Dependencies: flow.NewTaskIDs(prepareControlPlaneDeployments, waitUntilAPIServerIsDown),
+		})
+		scaleETCDToZero = g.Add(flow.Task{
```

Comment:
Agreed to Rafael's comment about first 5 steps. Making sure etcd is update not required. Cause we are not bringing up any apiserver or etcd dependent object in the flow.

This step `CreateETCDSnapshot` should also not required. Let etcd-druid handle this as a part of `scaleEtcdToZero`.

Reply:
Is that already part of the druid. Last time I spoke with @georgekuruvillak, he was about to implement it.